prov/efa/src/rxr/rxr_ep.c

```diff
@@ -375,6 +386,15 @@ void rxr_tx_entry_init(struct rxr_ep *ep, struct rxr_tx_entry *tx_entry,
 	else
 		memset(tx_entry->desc, 0, sizeof(tx_entry->desc));
 
+	/*
+	 * The prefix is currently not used by the sender, but needs to be
+	 * accounted for when copying the payload into the bounce-buffer.
+	 */ 
+	if (ep->use_zcpy_rx)
+		tx_entry->iov[0].iov_base = (char *)tx_entry->iov[0].iov_base
+					     + sizeof(struct rxr_pkt_entry)
+					     + sizeof(struct rxr_eager_msgrtm_hdr);
+
```

Comment:
Need to think about the possibility that caller provide multiple iov and the first iov's length is smaller than ```sizeof(struct rxr_pkt_entry) + sizeof(struct rxr_eager_msgrtm_hdr)```. I think there is function ```ofi_consume_iov``` we can use.

Reply:
The prefix element has to be atleast as big as msg_prefix_size so it won't spill over and I did not have a need for ` rxr_locate_iov_pos()`. We also don't want a memmove() that's in ofi_consume_iov(), so I left it as is. I can still use  rxr_locate_iov_pos, let me know if you prefer that over this arithmetic.

Reply:
I see. I think we just need an assertion here that ```tx_entry->iov[0].iov_len >= sizeof(struct rxr_pkt_entry)  + sizeof(struct rxr_eager_msgrtm_hdr)```

Also, do we need to subtract ```sizeof(struct rxr_pkt_entry)  + sizeof(struct rxr_eager_msgrtm_hdr)``` from  ```tx_entry->iov[0].iov_len``` here?

Reply:
Added the assertion in pkt_init for the recv, missed it here. Will add it.

The len would not change, the completion we write to the app will have to include the actual posted length, including the prefix.

Reply:
My worry is that during the initialization of an eager rtm, ```tx_entry->iov[0].iov_len``` was used to decide how many bytes are to be transferred over wire. If we change ```tx_entry->iov[0].iov_base``` but not ```tx_entry->iov[0].iov_len```, there will be invalid memory access.

Reply:
Done

Reply:
The data_size calculation in rxr_pkt_init_rtm would prevent that from happening.